lib/models/url_preferences_model.dart

```diff
@@ -1,18 +1,29 @@
-class UrlPreferences {
-  List<String> favouriteWidgetIds = [];
-  List<String> quarantineWidgetIds = [];
+import 'dart:convert';
 
-  UrlPreferences({this.favouriteWidgetIds, this.quarantineWidgetIds});
+import 'package:cogboardmobileapp/models/widget_model.dart';
 
-  factory UrlPreferences.fromJson(Map<String, dynamic> json) => UrlPreferences(
-        favouriteWidgetIds: (json['favouriteWidgetIds'] as List<dynamic>).cast<String>(),
-        quarantineWidgetIds: (json['quarantineWidgetIds'] as List<dynamic>).cast<String>(),
+class ConnectionPreferences {
+  List<DashboardWidget> favouriteWidgets = [];
+  List<DashboardWidget> quarantineWidgets = [];
+  String connectionName;
+  String connectionUrl;
+
+  ConnectionPreferences({this.favouriteWidgets, this.quarantineWidgets, this.connectionName, this.connectionUrl});
+
+  factory ConnectionPreferences.fromJson(Map<String, dynamic> json) => ConnectionPreferences(
+    favouriteWidgets: List<dynamic>.from(jsonDecode(json['favouriteWidgets'])).map((e) => DashboardWidget.fromJson(e)).toList(),
+    quarantineWidgets: List<dynamic>.from(jsonDecode(json['quarantineWidgets'])).map((e) => DashboardWidget.fromJson(e)).toList(),
```

Comment:
[LATER] extract method

Reply:
I have tried to do that but this is factory constructor
![image](https://user-images.githubusercontent.com/36708034/106283932-d1abe880-6242-11eb-8803-2751389a1e6e.png)
so there cant be invoked any methods apart from factory methods, List<dynamic>.from is a factory method
`external factory List.from(Iterable elements, {bool growable = true});`
DashboardWidget.fromJson(e) too is a factory method, so every method of extraction I tried failed, I would leave it as it is

Reply:
ok